oracle_server/oracle_server.go

```diff
@@ -479,8 +476,10 @@ func (os *OracleServer) doReport(curRoundCommitmentHash common.Hash, lastRoundDa
 	auth.GasTipCap = new(big.Int).SetUint64(os.gasTipCap)
 	auth.GasLimit = uint64(3000000)
 
-	// if there is no last round data, then we just submit the curRndCommitHash hash of current round.
-	if lastRoundData == nil {
+	// if there is no last round data or there were missing datapoint in last round data, then we just submit the
+	// commitment hash of current round as data might be available at current round. This vote will be reimbursed by the
+	// protocol, however it won't be abused as it is limited by the 1 vote per round rule.
+	if lastRoundData == nil || lastRoundData.MissingData {
```

Comment:
We are now submitting the round data report when we have `MissingData == false`, however we only set this to true on line 613 if the price is not included in the aggregated map of prices. What happens if a plugin returns a price of 0, without throwing any kind of error?

Reply:
If there is a price of 0 measured from market, that might means there is no liquidity at all, not sure if this is allowed to be reported. In this case, a TXN revertion might be expected. But I think we can check this before the reporting.

Reply:
I think since we are allowing independent implementations of plugins, this check is definitely good to have.